app/controllers/api/v1/accounts/credentials_controller.rb

```diff
@@ -22,4 +23,20 @@ def update
   def account_params
     params.permit(:display_name, :note, :avatar, :header, :locked)
   end
+
+  def user_settings_params
+    return nil unless params.key?(:source)
+
+    source_params = params.require(:source).permit(
+      :privacy,
```

Comment:
Why renaming `setting_default_privacy` as `privacy`? Indeed `setting` is kind of duplicate and fine to remove, but `default` is not such a duplicate and should be preserved.

Reply:
This is because that's the way it goes out in `verify_credentials` - the user of the api just modifies the values in the structure she receives and sends it back.